### PR TITLE
refactor UBI image dockerfiles to enable extensions

### PR DESCRIPTION
**Pull request by rplevka**
Adds some generic extension mechanism for users to include custom files and scripts in the container images.

Main motivation was to include promtail installation and configuration without having to hardcode it.

I've introduced 2 subdirs for this:
```
|
|- resources
    |- hostname--...x86_64.rpm
    |- startup.sh
|- setup_scripts
    |- 00_hostname.sh
```
The Dockerfiles will `ADD` the files to the build container and execute the scripts from `setup_scripts`.
I had to ADD these in 2 steps, as the `ADD` directive flattens the file structure. Having the setup scripts and other resources mixed together might lead to issues, if user would like to copy a shell script as a resource and not as a setup script.
One way to mitigate this might be to establish a pre||suffix for the setup scripts so we can pick them for an execution by a glob, but I find the separate directory method a little cleaner and more readable

All `UBI` dockerfiles already include `hostname` package installation, so I turned this step into an "extension" to demonstrate the intended workflow.

### CODE REVIEW

File: RHEL6/Dockerfile [hunk 1]
```diff
@@ -8,8 +8,13 @@ WORKDIR /root
 RUN sed -i -e 's/os.path.exists(HOST_CONFIG_DIR)/False/g' /usr/lib64/python2.6/site-packages/rhsm/config.py
 RUN echo "{\"virt.host_type\": \"Not Applicable\", \"virt.is_guest\": \"False\"}" > /etc/rhsm/facts/custom.facts
 
-ADD startup.sh /tmp/
-RUN chmod +x /tmp/startup.sh
+# add and process the external resources
+ADD resources/* /tmp/
+ADD setup_scripts/* /tmp/setup_scripts/
+RUN ls -l /tmp
```
Comment (ogajduse) [inline RHEL6/Dockerfile]: Is it a debug leftover?

File: UBI7/Dockerfile [hunk 2]
```diff
@@ -7,10 +7,13 @@ WORKDIR /root
...
 RUN echo "{\"virt.host_type\": \"Not Applicable\", \"virt.is_guest\": \"False\"}" > /etc/rhsm/facts/custom.facts
-ADD startup.sh /tmp/
-RUN chmod +x /tmp/startup.sh
-ADD hostname-3.13-3.el7.x86_64.rpm /tmp/
-RUN yum -y localinstall /tmp/hostname-3.13-3.el7.x86_64.rpm
+
+# add and process the external resources
+ADD resources/* /tmp/
+ADD setup_scripts/* /tmp/setup_scripts/
+WORKDIR /tmp
```
Comment (ogajduse) [inline UBI7/Dockerfile:15]: add a space between ADD and WORKDIR to keep it consistent with the other Dockerfiles.

File: RHEL6/Dockerfile [hunk 3]
```diff
@@ -8,8 +8,13 @@ WORKDIR /root
...
+WORKDIR /tmp
+RUN chmod +x *.sh setup_scripts/*.sh || true
```
Comment (ogajduse) [inline RHEL6/Dockerfile:16]: What are the cases where chmod fails? Does chmod behave differently on RHEL 6?

File: README.md [hunk 4]
```diff
@@ -13,6 +13,14 @@ or specify a directory and build the image manually
 
 ```docker build -t ch-d:sles11 SLES11/.```
 
+Certain dockerfiles (currently only `UBI`) support including of additional files and setup scripts by adding them to
```
Comment (ogajduse) [inline README.md]:
```suggestion
Certain dockerfiles (currently only `UBI` and RHEL6) support including of additional files and setup scripts by adding them to
```

**Review by ogajduse — COMMENTED** [review]
Nice!

File: RHEL6/Dockerfile (hunk 1 above)
Comment (rplevka) [reply RHEL6/Dockerfile]: indeed :( will remove

File: RHEL6/Dockerfile (hunk 3 above)
Comment (rplevka) [reply RHEL6/Dockerfile:16]:
```
$ chmod +x /tmp/fooo*.sh
chmod: cannot access '/tmp/fooo*.sh': No such file or directory.
```

I only address it here, as this one does not install `hostname` rpm by default, so the dirs are empty

**Comment by JacobCallahan** [comment]
I really like this addition! It gives a nice way to extend the container contents and setup. ACK pending @ogajduse's feedback

**Comment by rplevka** [comment]
@ogajduse addressed your comments, mind re-visiting?

**Review by ogajduse — APPROVED** [review]
Good job. LGTM

**Review by JacobCallahan — APPROVED** [review]

**Pull request merged**